src/Streamly/Internal/Data/Fold.hs

```diff
@@ -55,22 +52,25 @@ module Streamly.Internal.Data.Fold
     , rollingHashFirstN
     -- , rollingHashLastN
 
-    -- ** Fold Semigroups
-    , sconcat
-
-    -- ** Full Folds (Monoidal)
-    , mconcat
-    , foldMap
-    , foldMapM
-
-    -- ** Full Folds (To Containers)
+    -- ** Saturating Reducers
+    -- | 'product' terminates if it becomes 0. Other folds can theoretically
+    -- saturate on bounded types, and therefore terminate, however, they will
+    -- run forever on unbounded types like Integer/Double.
+    , sum
+    , product
+    , maximumBy
+    , maximum
+    , minimumBy
+    , minimum
 
+    -- ** Collectors
     , toList
     , toListRevF  -- experimental
+    , toParallelSVar
 
-    -- ** Partial Folds
+    -- * Terminating Folds
     , drainN
-    , drainSepBy
+    , drainWhile
```

Comment:
I renamed this because input is lost here. Maybe we should just remove this API. `drainSepBy` is ugly and I don't see the need for a function that drains even the seperator.

Reply:
I was thinking of using this when you anyway discard the rest of the stream. But that's not necessarily true. We can either remove it or move it to parsers.

Reply:
`drainSepBy` is a misleading name, it seems to indicate the condition is for `sepBy` but it is the opposite.

Reply:
We can move this to parsers.